code/game/objects/items/stacks/medical.dm

```diff
@@ -56,18 +56,13 @@
 				else if(src.heal_brute < 1)
 					user << "<span class='notice'> [src] won't help [M] at all.</span>"
 					return
-			user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")
-		else
-			var/t_himself = "itself"
-			if(user.gender == MALE)
-				t_himself = "himself"
-			else if(user.gender == FEMALE)
-				t_himself = "herself"
-			user.visible_message("<span class='notice'>[user] starts to apply [src] on [t_himself]...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
-			if(!do_mob(user, M, self_delay))	return
-			user.visible_message("<span class='green'>[user] applies [src] on [t_himself].</span>", "<span class='green'>You apply [src] on yourself.</span>")
-
-
+				else
+					var/mob/living/carbon/human/H = M
+					var/obj/item/organ/limb/affecting = H.get_organ(check_zone(user.zone_sel.selecting))
+					if(affecting.status != ORGAN_ORGANIC)
+						user << "<span class='notice'>Medicine won't work on a robotic limb!</span>"
+						return
+						user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")

```

Comment:
1) This entire block should be intended one level lower
2) You should make sure the mob is a human after you typecast it. Or even better, move it inside the ishuman(M) check right below.
3) Why are you displaying a visible_message() AFTER `return`ing? This is never going to get displayed. In fact, why is this even here? It looks like a botched indent but this would only trigger on healable simple animals! simple_animals with